swift_scality_backend/http_utils.py

```diff
@@ -112,6 +112,12 @@ def __init__(self, sock, debuglevel=0, strict=0, method=None,
             # possible.
             self.fp = SomewhatBufferedFileObject(sock, 'rb', 1024)
 
+        def fileno(self):
```

Comment:
I'd do this slightly differently. See this example:

```python
class A(object):
    def f(self):
        print 'a'

class B(A):
    if not hasattr(A, 'f'):
        def f(self):
            print 'b'

B().f()
```

Try changing `A.f` to `A.g` and re-run.